### PR TITLE
fix(e2e): drop httpbin.org from passthrough headers test, use real Anthropic

### PR DESCRIPTION
**Pull request by mubashir1osmani**
## Pre-Submission checklist

**Please complete all items before asking a LiteLLM maintainer to review your PR**

- [x] I have added meaningful tests
- [x] My PR passes all CI/CD checks (e.g., lint, format, unit tests)
- [x] My PR's scope is as isolated as possible; it only solves 1 specific problem
- [ ] I have received a Greptile **Confidence Score of at least 4/5** before requesting a maintainer review (Greptile reviews automatically once the PR is opened; only comment `@greptileai` to re-request a review after pushing changes)

## Delays in PR merge?

If you're seeing a delay in your PR being merged, ping the LiteLLM Team on [Slack (#pr-review)](https://join.slack.com/t/litellmossslack/shared_invite/zt-3o7nkuyfr-p_kbNJj8taRfXGgQI1~YyA).

## Screenshots / Proof of Fix

Verified live against a proxy running from this branch's commit (`15d8c6a938`) and the real Anthropic API, no mocks

Create the pass-through endpoint pointed at the real Anthropic Messages API with a static `x-api-key` the caller never sees:

```bash
curl -s -X POST http://localhost:4000/config/pass_through_endpoint \
  -H "Authorization: Bearer sk-1234" -H "Content-Type: application/json" \
  -d '{"path":"/e2e-passthrough-headers-test","target":"https://api.anthropic.com/v1/messages","headers":{"x-api-key":"os.environ/ANTHROPIC_API_KEY"}}'
# {"endpoints":[{"id":"ca49833f-5d4d-43d0-a079-683f6d390ced","path":"/e2e-passthrough-headers-test","target":"https://api.anthropic.com/v1/messages", ...}]}
```

Generate a scoped key, then call the route with a valid `x-pass-anthropic-version` and no client-supplied `x-api-key` at all - success proves the static header reached upstream:

```bash
curl -s -X POST http://localhost:4000/v1/messages... # via the custom path, key omitted for brevity
-H "x-pass-anthropic-version: 2023-06-01" -d '{"model":"claude-haiku-4-5-20251001","max_tokens":8,"messages":[{"role":"user","content":"Say hi."}]}'
# HTTP 200
# {"model":"claude-haiku-4-5-20251001","id":"msg_011CdFtui9dg1r6tkFjeG6KQ","type":"message","role":"assistant","content":[{"type":"text","text":"Hi! 👋 How can"}], ...}
```

Same call with a deliberately invalid `x-pass-anthropic-version` marker - Anthropic's 400 body echoes the exact marker verbatim, proving the `x-pass-` prefix was stripped and the caller's own header (not a cached or default one) reached upstream:

```bash
curl -s -X POST http://localhost:4000/v1/messages... \
-H "x-pass-anthropic-version: e2e-passhdr-d5923e3aef45" -d '{"model":"claude-haiku-4-5-20251001","max_tokens":8,"messages":[{"role":"user","content":"Say hi."}]}'
# HTTP 400
# {"type":"error","error":{"type":"invalid_request_error","message":"anthropic-version: \"e2e-passhdr-d5923e3aef45\" is not a valid version"},"request_id":"req_011CdFtunDTE9MFSKJq9W16d"}
```

## Type

🐛 Bug Fix
✅ Test

## Changes

The stage e2e failure was `httpbin.org` (an external echo service the test used as the pass-through target) returning a transient 503, unrelated to any litellm bug. Beyond the flakiness, the test also never exercised a real LLM provider

This swaps the target to the real Anthropic Messages API. Anthropic doesn't echo request headers back the way httpbin did, so the assertions had to change shape: a static `x-api-key` configured on the pass-through endpoint (the caller never supplies one) must reach upstream or the call 401s, and Anthropic's error body for an invalid `anthropic-version` echoes the exact value sent, which lets a unique-per-run marker sent via `x-pass-anthropic-version` prove the `x-pass-` stripping-and-forwarding mechanism attributably, without needing an echo service at all

## QA runbook

- tests/e2e/llm_translation/test_passthrough_headers_e2e.py::TestPassthroughHeaders::test_static_and_x_pass_headers_reach_upstream - a custom pass-through endpoint's static header reaches the real Anthropic API, and a client's `x-pass-*` header is forwarded with the prefix stripped
  - [x] Create a pass-through endpoint: `curl -X POST http://localhost:4000/config/pass_through_endpoint -H "Authorization: Bearer sk-1234" -d '{"path":"/anthropic-passthrough-demo","target":"https://api.anthropic.com/v1/messages","headers":{"x-api-key":"os.environ/ANTHROPIC_API_KEY"}}'`
  - [x] Generate a key scoped to that route: `curl -X POST http://localhost:4000/key/generate -H "Authorization: Bearer sk-1234" -d '{"models":[],"allowed_passthrough_routes":["/anthropic-passthrough-demo"]}'`
  - [x] Call the route with the key and `x-pass-anthropic-version: 2023-06-01`, plus a minimal Anthropic Messages body - expect a 200 with real assistant content
  - [x] Call it again with `x-pass-anthropic-version: some-garbage-value` - expect a 400 whose body contains the exact string `some-garbage-value`
  - [x] Sanity check: this test makes sense to add and is not hand-wavey (e.g., assert actual expected spend instead of just spend > 0) or potentially flaky

### Final Attestation

- [x] The tests check the right things, including the edge cases, and regressions in the respective real-world customer use-cases are not possible after this PR

### CODE REVIEW

File: tests/e2e/llm_translation/test_passthrough_headers_e2e.py [hunk 1]
```diff
@@ -128,23 +121,32 @@ def test_static_and_x_pass_headers_reach_upstream(
...
-        assert upstream.get(PASS_HEADER_STEM) == client_value, (
-            f"x-pass-* header should strip the prefix and forward as {PASS_HEADER_STEM!r}; "
-            f"got {upstream}"
+        assert blocked.status_code == 400, (
+            f"expected Anthropic to reject the invalid anthropic-version, got "
+            f"{blocked.status_code}: {blocked.body[:300]}"
         )
-        assert PASS_HEADER_NAME not in upstream, (
-            "upstream must not see the x-pass- prefix; proxy should strip it"
+        assert invalid_version in blocked.body, (
```
[the remark this answers is not in the document]
Comment (mubashir1osmani) [reply tests/e2e/llm_translation/test_passthrough_headers_e2e.py:149]: Good catch on the coverage difference from the echo upstream. Two things make this safe on this endpoint as written.

The marker-echo assertion already proves the prefix was stripped, not just that some header arrived. Anthropic only echoes the value because it was received under the name anthropic-version. If the proxy had forwarded the raw x-pass-anthropic-version instead of stripping it, Anthropic would see no version and return a 400 about a missing anthropic-version, whose body would not contain our per-run marker, so this assertion would fail.

The one shape a real upstream cannot distinguish is the proxy forwarding both the stripped anthropic-version and the raw x-pass-anthropic-version. That cannot arise here: the endpoint leaves forward_headers off, and forward_headers_from_request only adds the stripped header in that mode; it never merges the raw request headers (litellm/passthrough/utils.py). The additive raw-header merge only happens under forward_headers=True, which this endpoint does not set, so no test change is needed.

**Review by yucheng-berri — APPROVED** [review]
